services/src/main/java/org/jboss/windup/web/services/service/MigrationProjectService.java

```diff
@@ -96,6 +96,25 @@ public MigrationProject createProject(MigrationProject project)
         return project;
     }
 
+    /**
+     * Delete old projects created during wizard but not finished.
+     *
+     * @param olderThanMinutes The limit to consider provisional projects old.
+     */
+    @Transactional
+    public void deleteOldProvisionalProjects(int olderThanMinutes)
+    {
+        String query = "SELECT pr FROM " + MigrationProject.class.getSimpleName() + " pr WHERE pr.provisional = TRUE ";
+        if (olderThanMinutes > 0)
+            query += " AND pr.lastModified < DATEADD('MINUTE', -"+olderThanMinutes+", CURRENT_TIMESTAMP) "; // JPA had issues with a param here.
+        List<MigrationProject> provisional = entityManager.createQuery(query, MigrationProject.class).getResultList();
+
+        LOG.info("Deleting " + provisional.size() + " provisional projects older than "+olderThanMinutes+" minutes.");
+        provisional.forEach(project -> {
+            this.deleteProject(project);
```

Comment:
(Oh, I guess I got it now. You need it to delete underlying applications and so on...). Ignore previous comment ;)